### PR TITLE
Drop `internal/xnet` + replace permanent connection into newest connection each discovery

### PR DESCRIPTION
**Pull request by asmyasnikov**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-
-
-

### CODE REVIEW

File: internal/balancer/balancer.go [hunk 1]
```diff
@@ -71,7 +65,12 @@ func (b *Balancer) clusterDiscovery(ctx context.Context) (err error) {
 		)
 	}()
 
-	endpoints, err = b.discovery.Discover(ctx)
+	client := b.discoveryClient()
+	if err != nil {
```
Comment (TheHurd) [inline internal/balancer/balancer.go:69]: what error are we checking in this place? (where did you get it from?)

File: internal/discovery/discovery.go [hunk 2]
```diff
@@ -20,24 +20,31 @@ import (
...
-	config  config.Config
-	service Ydb_Discovery_V1.DiscoveryServiceClient
-	cc      conn.Conn
+	config      config.Config
+	grpcOptions []grpc.DialOption
+}
+
+func (c *Client) client(ctx context.Context) (Ydb_Discovery_V1.DiscoveryServiceClient, *grpc.ClientConn, error) {
+	cc, err := grpc.DialContext(ctx,
+		"dns:///"+c.config.Endpoint(),
```
Comment (TheHurd) [inline internal/discovery/discovery.go:41]: it is better to do such things once, when parsing the configuration, and not every time you connect

File: internal/discovery/discovery.go [hunk 3]
```diff
@@ -65,7 +72,15 @@ func (c *Client) Discover(ctx context.Context) (endpoints []endpoint.Endpoint, e
 		return nil, xerrors.WithStackTrace(err)
 	}
 
-	response, err = c.service.ListEndpoints(ctx, &request)
+	client, cc, err := c.client(ctx)
+	if err != nil {
+		return nil, xerrors.WithStackTrace(err)
+	}
+	defer func() {
+		_ = cc.Close()
```
Comment (TheHurd) [inline internal/discovery/discovery.go:80]: here and in other similar places maybe it's better to make a wrapper for close() to apply a simpler defer form?

File: internal/balancer/balancer.go (hunk 1 above)
Comment (asmyasnikov) [reply internal/balancer/balancer.go:69]: ack

File: internal/discovery/discovery.go (hunk 3 above)
Comment (asmyasnikov) [reply internal/discovery/discovery.go:80]: simple form have linter issue (errcheck)

Comment (hurd54) [reply internal/discovery/discovery.go:80]: that's why I suggested making a wrapper for close()
something like:

```go
func (c someType) silentClose {
  _ = c.Close()
}

...
// somewhere in the code
defer cc.silentClose()
```

**Pull request merged**